#### Pralay K Kalaskar
#### (pkkdkalaskar71@gmail.com)
#### +91-7066216365
#### www.linkedin.com/in/pralay-kalaskar-1010
#### Task 04 :- 27-july-2024

## used "Student_performance_data.csv" file

In [31]:
# Importing libraries
import pandas as pd
import numpy as np

In [32]:
# loading dataset into a Pandas Dataframe
d = pd.read_csv("Student_performance_data.csv")

In [35]:
# "describe()" method is used to generate descriptive statistics of DataFrames or Series.
# It provides a summary of the data, which is essential for understanding its distribution and identifying patterns or anomalies.
d.describe()

StudentID          Age       Gender    Ethnicity  ParentalEducation  \
count  2392.000000  2392.000000  2392.000000  2392.000000        2392.000000   
mean   2196.500000    16.468645     0.510870     0.877508           1.746237   
std     690.655244     1.123798     0.499986     1.028476           1.000411   
min    1001.000000    15.000000     0.000000     0.000000           0.000000   
25%    1598.750000    15.000000     0.000000     0.000000           1.000000   
50%    2196.500000    16.000000     1.000000     0.000000           2.000000   
75%    2794.250000    17.000000     1.000000     2.000000           2.000000   
max    3392.000000    18.000000     1.000000     3.000000           4.000000   

       StudyTimeWeekly     Absences     Tutoring  ParentalSupport  \
count      2392.000000  2392.000000  2392.000000      2392.000000   
mean          9.771992    14.541388     0.301421         2.122074   
std           5.652774     8.467417     0.458971         1.122813   
min           0.001057     0.000000     0.000000         0.000000   
25%           5.043079     7.000000     0.000000         1.000000   
50%           9.705363    15.000000     0.000000         2.000000   
75%          14.408410    22.000000     1.000000         3.000000   
max          19.978094    29.000000     1.000000         4.000000   

       Extracurricular       Sports        Music  Volunteering          GPA  \
count      2392.000000  2392.000000  2392.000000   2392.000000  2392.000000   
mean          0.383361     0.303512     0.196906      0.157191     1.906186   
std           0.486307     0.459870     0.397744      0.364057     0.915156   
min           0.000000     0.000000     0.000000      0.000000     0.000000   
25%           0.000000     0.000000     0.000000      0.000000     1.174803   
50%           0.000000     0.000000     0.000000      0.000000     1.893393   
75%           1.000000     1.000000     0.000000      0.000000     2.622216   
max           1.000000     1.000000     1.000000      1.000000     4.000000   

        GradeClass  
count  2392.000000  
mean      2.983696  
std       1.233908  
min       0.000000  
25%       2.000000  
50%       4.000000  
75%       4.000000  
max       4.000000

### The "describe()" method is a powerful tool for quickly obtaining a summary of the data's statistical properties, helping with exploratory data analysis and understanding the distribution of the dataset.
### It adapts to different data types and provides essential statistics to inform further data analysis and preprocessing.

### In Python, specifically within the pandas library, the "describe()" method is used to generate descriptive statistics of DataFrames or Series.
### It provides a summary of the data, which is essential for understanding its distribution and identifying patterns or anomalies.
### Here’s a breakdown of what describe() does, Key Features of "describe()"
######
### [1]. Summarizes Data :- "describe()" calculates and returns summary statistics for numeric columns by default. For categorical or object-type columns, it provides different statistics.

### [2]. Default Statistics :-
### (A) :- For numeric columns, "describe()" typically provides the following statistics :-
##### 1. count :- Number of non-null entries.
##### 2. mean :- Average of the values.
##### 3. std :- Standard deviation of the values.
##### 4. min :- Minimum value.
##### 5. 25% :- 25th percentile (first quartile).
##### 6. 50% :- 50th percentile (median or second quartile).
##### 7. 75% :- 75th percentile (third quartile).
##### 8. max :- Maximum value.

### (B) :- For categorical columns, it provides :-
##### 1. count :- Number of non-null entries.
##### 2. unique :- Number of unique values.
##### 3. top :- Most frequent value.
##### 4. freq :- Frequency of the most frequent value.

### [3]. Handling Different Data Types :- By default, "describe()" includes numeric data. If specified, it can include other data types (such as objects and categories) or exclude them.

### [4]. Percentiles :- You can customize which percentiles to include in the output using the percentiles parameter.

### [5]. Datetime Handling :- When dealing with datetime data, "describe()" can treat datetime columns as numeric if specified with the datetime_is_numeric parameter.

## Report
######
##### [A] Data Columns :- names and description.
01. StudentID :- Unique identifier for each student.
02. Age :- Age of the students.
03. Gender :- Gender of the students, where 0 might represent one gender (e.g., male) and 1 another (e.g., female).
04. Ethnicity :- Ethnic background of the students, with codes representing different ethnic groups.
05. ParentalEducation :- Level of parental education, with numerical codes.
06. StudyTimeWeekly :- Number of hours the student studies per week.
07. Absences :- Number of school absences.
08. Tutoring :- Whether the student receives tutoring, possibly binary (0 or 1).
09. ParentalSupport :- Level of support from parents, likely binary (0 or 1).
10. Extracurricular :- Number of extracurricular activities the student participates in.
11. Sports :- Whether the student participates in sports, likely binary (0 or 1).
12. Music :- Whether the student participates in music activities, likely binary (0 or 1).
13. Volunteering :- Whether the student volunteers, likely binary (0 or 1).
14. GPA :- Grade Point Average of the student.
15. GradeClass :- The class or grade level of the student.
######
##### [B] Summary Statistics :- 
01. count :- Number of data points for each attribute (2392 in this case).
02. mean :- Average value of the attribute across all students.
03. std :- Standard deviation, which measures the amount of variation or dispersion of the attribute.
04. min :- Minimum value observed for the attribute.
05. 25% :- 25th percentile, meaning 25% of the data points are below this value.
06. 50% :- Median (or 50th percentile), meaning 50% of the data points are below this value.
07. 75% :- 75th percentile, meaning 75% of the data points are below this value.
08. max :- Maximum value observed for the attribute.
######
##### This summary provides insights into the distribution and characteristics of the student population in the dataset.
######
### For all 15 columns details, the explaination is in respective word file of each column name.
### OR (*Optional*) you can call them by using the below code. Just change the name of file with column name.

In [36]:
# Importing libraries
from docx import Document

# Loading the document
doc = Document('01_StudentID.docx')  # you can change the file name here in this section

# Read and print each paragraph
for para in doc.paragraphs:
    print(para.text)

Statistical overview of the “StudentID” column from the dataset. Here’s what each value means:
Count (2392): This tells you the number of student IDs in the dataset. Here, there are 2392 individual IDs.
Mean (2196.5): This is the average student ID number. It’s calculated by adding up all the student IDs and dividing by the total number of IDs. On average, the student ID is 2196.5.
Standard Deviation (690.6552444): This measures the amount of variation or dispersion in the student IDs. A standard deviation of approximately 690.66 means that student IDs tend to deviate from the mean by about this amount. A larger standard deviation indicates more spread out IDs.
Min (1001): This is the smallest student ID number in the dataset. The lowest ID recorded is 1001.
25th Percentile (1598.75): Also known as the first quartile. It means that 25% of the student IDs are less than or equal to 1598.75. Essentially, this value marks the lower 25% of the dataset.
50th Percentile (2196.5): This is the 